src/cli.rs

```diff
@@ -51,35 +55,21 @@ struct Args {
 }
 
 #[instrument(level = "debug")]
-fn endpoints_parser(endpoint: &str) -> Result<String, String> {
-    let endpoint = endpoint.to_lowercase();
-
-    let mut split = endpoint.split(':');
-    if split.clone().count() != 2 && split.clone().count() != 3 {
-        return Err("Wrong endpoint format".to_string());
-    }
-
-    let kind = split.next().expect(
-        "Endpoint should start with one of the kinds: file, udps, udpc, udpb, tcps, tcpc, or serial",
-    );
-    if !matches!(
-        kind,
-        "file" | "udps" | "udpc" | "udpb" | "tcps" | "tcpc" | "serial"
-    ) {
-        return Err(format!("Unknown kind: {kind:?} for endpoint"));
-    }
-
-    // Add your custom validation logic here if needed
-    Ok(endpoint)
+fn endpoints_parser(endpoint: &str) -> Result<Arc<dyn drivers::Driver>, String> {
+    drivers::endpoints()
+        .iter()
+        .map(|driver| driver.driver_ext.create_endpoint(endpoint))
+        .filter_map(|driver| driver)
+        .next()
+        .ok_or_else(|| format!("Invalid endpoint: {}", endpoint))
 }
 
-#[derive(Debug)]
 struct Manager {
-    clap_matches: Args,
+    pub clap_matches: Args,
```

Comment:
```suggestion
    clap_matches: Args,
```
this probably doesn't need to be public